__tests__/app.test.js

```diff
@@ -175,3 +166,81 @@ describe("GET/api/articles", () => {
       });
   });
 });
+
+describe("GET/api/articles/:article_id/comments", () => {
+  test("should return a status code 200", () => {
+    return request(app).get("/api/articles/1/comments").expect(200);
+  });
+  test("should return an array of comments for the given article", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.articles.length).toBe(11);
+        response.body.articles.forEach((article) => {
+          expect(article).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: 1,
+          });
+        });
+      });
+  });
+  test("should return an array of comments for the given article ordered by date created descending", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.articles).toBeSortedBy("created_at", {
+          descending: true,
+        });
+      });
+  });
+  test('should return an array of comments for the given article ordered by votes descending', () => {
+    return request(app)
+      .get("/api/articles/1/comments?sort_by=votes&&order=DESC")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.articles).toBeSortedBy("votes", {
+            descending: true
+        })
+      })
+  });
+  test('should return an array of comments for the given article ordered by author alphabetically', () => {
+    return request(app)
+      .get("/api/articles/1/comments?sort_by=author&&order=ASC")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.articles).toBeSortedBy("author", {
+            descending: false
+        })
+      })
+  });
+  test('should return appropriate error when passed a valid but non existant article ID ', () => {
+    return request(app)
+    .get("/api/articles/999/comments")
+    .expect(404)
+    .then((response) => {
+        expect(response.body.msg).toBe("not found")
+    })
+  });
+  test('should return an appropriate error when passed an invalid sort_by query', () => {
+    return request(app)
+    .get("/api/articles/1/comments?sort_by=id")
+    .expect(400)
+    .then((response) => {
+        expect(response.body.msg).toBe("bad request")
+    })
+  });
+  test('should return an appropriate error when passed an invalid order query', () => {
+    return request(app)
+    .get("/api/articles/1/comments?order=lowest")
+    .expect(400)
+    .then((response) => {
+        expect(response.body.msg).toBe("bad request")
+    })
```

Comment:
These last 2 tests aren't needed either as mentioned above> there is a test missing though. What if the article does exist but has no comments associated with it? Like article id 13? This should respond with a 200 status and an empty array. Please do add this test and any logic needed